src/main/java/io/github/thebusybiscuit/slimefun4/core/services/holograms/HologramsService.java

```diff
@@ -0,0 +1,311 @@
+     * 
+     * @return Whether this could be a hologram
+     */
+    private boolean isHologram(@Nonnull Entity n) {
+        if (n instanceof ArmorStand) {
+            ArmorStand armorstand = (ArmorStand) n;
+
+            // The absolute minimum requirements to count as a hologram
+            return !armorstand.isVisible() && armorstand.isSilent() && !armorstand.hasGravity();
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * This will cast the {@link Entity} to an {@link ArmorStand} and it will apply
+     * all necessary attributes to the {@link ArmorStand}, then return a {@link Hologram}.
+     * 
+     * @param position
+     *            The {@link BlockPosition} of this hologram
+     * @param entity
+     *            The {@link Entity}
+     * @param container
+     *            The {@link PersistentDataContainer} of the given {@link Entity}
+     * 
+     * @return The {@link Hologram}
+     */
+    @Nullable
+    private Hologram getAsHologram(@Nonnull BlockPosition position, @Nonnull Entity entity, @Nonnull PersistentDataContainer container) {
+        if (entity instanceof ArmorStand) {
+            ArmorStand armorstand = (ArmorStand) entity;
+
+            armorstand.setVisible(false);
+            armorstand.setInvulnerable(true);
+            armorstand.setSilent(true);
+            armorstand.setMarker(true);
+            armorstand.setAI(false);
+            armorstand.setGravity(false);
+            armorstand.setRemoveWhenFarAway(false);
+
+            // Set a persistent tag to re-identify the correct hologram later
+            container.set(persistentDataKey, PersistentDataType.LONG, position.getPosition());
+
+            // Store in cache for faster access
+            Hologram hologram = new Hologram(armorstand.getUniqueId());
+            cache.put(position, hologram);
+
+            return hologram;
+        } else {
+            // This should never be reached
+            return null;
+        }
+    }
+
+    /**
+     * This updates the {@link Hologram}.
+     * You can use it to set the nametag or other properties.
+     * <p>
+     * <strong>This method must be executed on the main {@link Server} {@link Thread}.</strong>
+     * 
+     * @param loc
+     *            The {@link Location}
+     * @param consumer
+     *            The callback to run
+     */
+    private void updateHologram(@Nonnull Location loc, @Nonnull Consumer<Hologram> consumer) {
+        Validate.notNull(loc, "Location must not be null");
+        Validate.notNull(consumer, "Callbacks must not be null");
+
+        Runnable runnable = () -> {
+            try {
+                Hologram hologram = getHologram(loc, true);
+
+                if (hologram != null) {
+                    consumer.accept(hologram);
+                }
+            } catch (Exception | LinkageError x) {
+                SlimefunPlugin.logger().log(Level.SEVERE, "Something went wrong while trying to update a hologram", x);
```

Comment:
Should we add more info here? Like the location where this happened. I am not sure whether the error already has it and if it is even useful at all but if it's nowhere more information is never bad

Reply:
In theory, but the catch-block shouldn't ever be reached really, it's just an in-case-of-emergency scenario anyway.
And if it is, then a threading issue or a NPE is probably the most likeliest event.
The Location would not be needed in this case, as any exception would happen on a higher level anywhere.
But as you said, more context is never a bad thing.

Reply:
Added the location.